src/zarr/storage/_obstore.py

```diff
@@ -224,7 +224,7 @@ async def set_if_not_exists(self, key: str, value: Buffer) -> None:
         self._check_writable()
         buf = value.to_bytes()
         with contextlib.suppress(obs.exceptions.AlreadyExistsError):
-            await obs.put_async(self.store, key, buf, mode="Create")
+            await obs.put_async(self.store, key, buf, mode="Create", conditional_put="etag")
```

Comment:
I didn't see this last commit; you had to set `conditional_put="etag"` on the _store_, not on the `put_async` call.